adapters/pubmatic/pubmatic.go

```diff
@@ -428,25 +427,25 @@ func validateAdSlot(adslot string, imp *openrtb.Imp) error {
 	return nil
 }
 
-func assignBannerSize(banner *openrtb.Banner) error {
-	if banner == nil {
-		return nil
-	}
-
+func assignBannerSize(banner *openrtb.Banner) (*openrtb.Banner, error) {
 	if banner.W != nil && banner.H != nil {
-		return nil
+		return banner, nil
 	}
 
 	if len(banner.Format) == 0 {
-		return errors.New(fmt.Sprintf("No sizes provided for Banner %v", banner.Format))
+		return banner, errors.New(fmt.Sprintf("No sizes provided for Banner %v", banner.Format))
```

Comment:
I think this should return a `nil` banner pointer if there's an error.

Reply:
Good catch. Removed